pymc3/ode/ode.py

```diff
@@ -1,211 +1,222 @@
+    ]
+    __props__ = ("func", "times", "n_states", "n_theta", "t0")
+
+    def __init__(self, func, times, *, n_states, n_theta, t0=0):
+        if not callable(func):
+            raise ValueError("Argument func must be callable.")
+        if n_states < 1:
+            raise ValueError("Argument n_states must be at least 1.")
+        if n_theta <= 0:
+            raise ValueError("Argument n_theta must be positive.")
+
+        # Public
+        self.func = func
+        self.t0 = t0
+        self.times = tuple(times)
+        self.n_times = len(times)
+        self.n_states = n_states
+        self.n_theta = n_theta
+        self.n_p = n_states + n_theta
+
+        # Private
+        self._augmented_times = np.insert(times, 0, t0).astype(floatX)
+        self._augmented_func = augment_system(func, self.n_states, self.n_p)
+        self._sens_ic = self._make_sens_ic()
+
+        # Cache symbolic sensitivities by the hash of inputs
+        self._apply_nodes = {}
+        self._output_sensitivities = {}
+    
+    def _make_sens_ic(self):
+        """
+        The sensitivity matrix will always have consistent form. (n_states, n_states + n_theta)
+
+        If the first n_states entries of the parameters vector in the simulate call
+        correspond to initial conditions of the system,
+        then the first n_states columns of the sensitivity matrix should form
+        an identity matrix.
+
+        If the last n_theta entries of the parameters vector in the simulate call
+        correspond to ode paramaters, then the last n_theta columns in
+        the sensitivity matrix will be 0.        
+        """
+
+        # Initialize the sensitivity matrix to be 0 everywhere
+        sens_matrix = np.zeros((self.n_states, self.n_states + self.n_theta), dtype=floatX)
+
+        # Slip in the identity matrix in the appropirate place
+        sens_matrix[:,:self.n_states] = np.eye(self.n_states, dtype=floatX)
+
+        # We need the sensitivity matrix to be a vector (see augmented_function)
+        # Ravel and return
+        dydp = sens_matrix.ravel()
+        return dydp
+
+    def _system(self, Y, t, p):
+        """This is the function that will be passed to odeint. Solves both ODE and sensitivities.
+
+        Args:
+            Y: augmented state vector (n_states + n_states + n_theta)
+            t: current time
+            p: parameter vector (y0, theta)
+        """
+        dydt, ddt_dydp = self._augmented_func(Y[:self.n_states], t, p, Y[self.n_states:])
+        derivatives = np.concatenate([dydt, ddt_dydp])
+        return derivatives
+
+    def _simulate(self, y0, theta):
+        # Initial condition comprised of state initial conditions and raveled sensitivity matrix
+        s0 = np.concatenate([y0, self._sens_ic])
+
+        # perform the integration
+        sol = scipy.integrate.odeint(
+            func=self._system, y0=s0, t=self._augmented_times, args=(np.concatenate([y0, theta]),)
```

Comment:
is this `np.concatenate([y0, theta])` instead of `np.concatenate([theta, y0])` what broke things?

Reply:
Yes. In my refactor, I changed the argument-order for the Op, but did not change it for the inner workings (augment_system).
The gradient however has entries in the same order as the Op inputs, whereas the column order of the sensitivity matrix is determined by those inner workings...